linera-service/src/node_service.rs

```diff
@@ -950,53 +886,24 @@ where
         Ok(())
     }
 
-    /// Handles queries for user applications.
-    async fn user_application_query(
-        &self,
-        application_id: UserApplicationId,
-        request: &Request,
-        chain_id: ChainId,
-    ) -> Result<async_graphql::Response, NodeServiceError> {
-        let QueryOutcome {
-            response: user_response_bytes,
-            operations,
-        } = self
-            .query_user_application(application_id, request, chain_id)
-            .await?;
-
-        ensure!(
-            operations.is_empty(),
-            NodeServiceError::UnexpectedOperationsFromQuery
-        );
-
-        Ok(serde_json::from_slice(&user_response_bytes)?)
-    }
-
-    /// Handles mutations for user applications.
-    async fn user_application_mutation(
+    /// Handles service queries for user applications (including mutations).
+    async fn handle_service_request(
         &self,
         application_id: UserApplicationId,
-        request: &Request,
+        request: Vec<u8>,
         chain_id: ChainId,
-    ) -> Result<async_graphql::Response, NodeServiceError> {
-        debug!("Request: {:?}", &request);
+    ) -> Result<Vec<u8>, NodeServiceError> {
         let QueryOutcome {
             response,
             operations,
         } = self
             .query_user_application(application_id, request, chain_id)
             .await?;
-        let graphql_response = serde_json::from_slice::<async_graphql::Response>(&response)?;
-        if graphql_response.is_err() {
-            let errors = graphql_response
-                .errors
-                .iter()
-                .map(|e| e.to_string())
-                .collect();
-            return Err(NodeServiceError::ApplicationServiceError { errors });
+        if operations.is_empty() {
+            return Ok(response);
         }
-        trace!("Operations: {operations:?}");
 
+        trace!("Query requested a new block with operations: {operations:?}");
```

Comment:
Maybe there should be a way for the user to tell the client that it does _not_ want to create a new block, regardless of what the application returns.

Reply:
Yes we're certainly not going to auto-sign block proposals like this in the future but instead properly integrate with wallets.